File: sparkJava/src/main/java/com/beifeng/core/Persist.java
package com.beifeng.core;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

/**
 * RDD持久化
 * time: 2018-08-12 13:19
 */
public class Persist {

    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
                .setAppName("Persist")
                .setMaster("local");

        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> lines = sc.textFile("D:\\spark.txt").cache();
        long beginTime = System.currentTimeMillis();
        long count = lines.count();
        System.out.println(count);
        long endTime = System.currentTimeMillis();
        System.out.println("cost " + (endTime - beginTime) + " milliseconds");


        beginTime = System.currentTimeMillis();
        count = lines.count();
        System.out.println(count);
        endTime = System.currentTimeMillis();
        System.out.println("cost " + (endTime - beginTime) + " milliseconds");

        // cost 395 milliseconds cost 87 milliseconds no cache
        // cost 447 milliseconds cost 28 milliseconds cache
    }
}

File: sparkJava/src/main/java/com/beifeng/scala/core/Persist.scala
package com.beifeng.scala.core

import org.apache.spark.{SparkConf, SparkContext}

object Persist {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
      .setAppName("Persist")
      .setMaster("local")

    val sc = new SparkContext(conf)

    val lines = sc.textFile("D:\\sparkTestFile\\spark.txt").cache()
    var beginTime = System.currentTimeMillis()
    var count = lines.count()
    println(count)
    var endTime = System.currentTimeMillis()
    println("cost " + (endTime - beginTime) + " milliseconds")


    beginTime = System.currentTimeMillis()
    count = lines.count()
    println(count)
    endTime = System.currentTimeMillis()
    println("cost " + (endTime - beginTime) + " milliseconds")
  }
}
